wcomponents-core/src/main/java/com/github/bordertech/wcomponents/WMessageBox.java

```diff
@@ -164,8 +166,10 @@ public Type getType() {
 	 * @param args optional arguments for the message format string.
 	 */
 	public void setTitleText(final String title, final Serializable... args) {
-		MessageModel model = getOrCreateComponentModel();
-		model.title = I18nUtilities.asMessage(title, args);
+		if (getTitleText() == null || !getTitleText().equals(title)) {
```

Comment:
Is there a need to handle the case where `getTitleText() == null` *and* arg `title` is also `null`? Pointless, maybe, but it is a public member so it is notionally possible that an external call is trying to reset the title (by making it null) _just in case_ it has been set when it has not yet been set. Since the aim of this code is to remove the creation of unnecessary models all cases should be handled to avoid the unnecessary model.

Reply:
To address the problem of both arg and getArg being null when they are Strings (my reason for not using != above), I have used the following to all your other comments regarding creating a model when both are null:
```
if ((title != null && getTitleText() == null)
	|| (title == null && getTitleText() != null)
	|| (getTitleText() != null && !getTitleText().equals(title))) {
        getOrCreateComponentModel.title = title (or whatever)
```